engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/SnapshotStrategyBase.java

```diff
@@ -21,8 +21,11 @@
 import org.apache.cloudstack.engine.subsystem.api.storage.SnapshotInfo;
 import org.apache.cloudstack.engine.subsystem.api.storage.SnapshotService;
 import org.apache.cloudstack.engine.subsystem.api.storage.SnapshotStrategy;
+import org.apache.cloudstack.framework.config.ConfigKey;
 
-public abstract class SnapshotStrategyBase implements SnapshotStrategy {
+import com.cloud.utils.component.ManagerBase;
+
+public abstract class SnapshotStrategyBase extends ManagerBase implements SnapshotStrategy {
```

Comment:
SnapshotStrategyBase  extends ManagerBase, since it contains configure(), in which config params and threadpools can be set/initialised.